text/0063-sdk-crash-monitoring.md

```diff
@@ -0,0 +1,81 @@
+# SDK Crash Monitoring
+
+- Start Date: 2023-01-18
+- RFC Type: feature
+- RFC PR: [#33](https://github.com/getsentry/rfcs/pull/33)
+- RFC Status: draft
+- RFC Driver: [Philipp Hofmann](https://github.com/philipphofmann)
+- RFC Approver: tbd
+
+## Summary
+
+This RFC aims to detect crashes caused by our SDKs to improve reliability.
+
+## Motivation
+
+As an APM company, the reliability of our SDKs is one of our most essential quality goals. If our SDK breaks the customer, we fail. Our SDK philosophy refers to this as [degrade gracefully](https://develop.sentry.dev/sdk/philosophy/#degrade-gracefully).
+
+For some SDKs, like mobile SDKs, we primarily rely on users to reach out when our SDKs cause crashes cause we don't operate them in production. If users don't report them, we are unaware. Instead, we should detect crashes caused by our SDKs when they happen so we can proactively fix them.
+
+## Background
+
+The Google Play SDK Console provides insights into crashes for SDK maintainers. We regularly use it for the Android/Java SDK. While it would be great also to build something similar for SDK maintainers within Sentry, it's a bit complicated cause of PII and such. Narrowing down the scope to only Sentry SDKs makes the problem easier to solve.
+
+## Options Considered
+
+- [Option 1: SDKs report FIOMT as errors](#option-1)
+- [Option 2: SDKs report FIOMT as performance issues](#option-2)
+- [Option 3: Ingest reports FIOMT as performance issues](#option-3)
+
+### Option 1: Detect during event processing <a name="option-1"></a>
+
+During event processing, after processing the stacktrace, the server detects if a crash stems from any of our SDKs by looking at the top frames of the stacktrace. If the server detects that it does, it could duplicate the event and store it in a special-cased sentry org where each SDK gets its project.
```

Comment:
Can we maintain data locality with this approach (ie. the hybrid cloud project)?

Reply:
Excellent point, @mdtro. That's a problem to solve. @jjbayer, can you maybe answer that question ⬆️ ?

Reply:
I guess it could be solved by actually posting HTTP requests to `sentry.sentry.io` to create the new issues (i.e. all derived errors go to the same region), but that might be problematic for performance. We can always heavily sample the number of derived errors we send though.

Reply:
@mdtr @jjbayer, why do we need data locality?